copylot/hardware/lights/voltran/voltran.py

```diff
@@ -0,0 +1,516 @@
+"""
+@author:edyoshikun
+
+Voltran Stradus Laser python wrapper using RS-232 -> COM device. 
+
+For more details regarding operation, refer to the manuals in https://www.vortranlaser.com/
+"""
+from copylot.hardware.lights.abstract_light import AbstractLight
+import serial
+from serial import SerialException
+from serial.tools import list_ports
+import time
+import logging
+
+# Setting up the Logger
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+handler = logging.StreamHandler()
+handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+# --------------------------------------------------------------------------------------------------
+
+# ---------------------------------------CONSTANTS-------------------------------------------------
+
+# TODO: this probably is better if it's a dictionary and make
+GLOBAL_CMD = ['ECHO', 'PROMPT']
+GLOBAL_QUERY = ['?BPT', '?H', '?IL', '?SFV', '?SPV']
+LASER_CMD = ['C', 'DELAY', 'EPC', 'LC', 'LE', 'LP', 'PP', 'PUP']
+LASER_QUERY = [
+    '?C',
+    '?CC',
+    '?CT',
+    '?DELAY',
+    '?EPC',
+    '?FC',
+    '?FD',
+    '?FP',
+    '?FV',
+    '?LC',
+    '?LCS',
+    '?LE',
+    '?LH',
+    '?LI',
+    '?LP',
+    '?LPS',
+    '?LW',
+    '?MAXP',
+    '?OBT',
+    '?OBTS',
+    '?PP',
+    '?PUL',
+    '?RP',
+]
+VOLTRAN_CMDS = GLOBAL_CMD + GLOBAL_QUERY + LASER_CMD + LASER_QUERY
+
+
+def get_lasers():
+    com_ports = list_ports.comports()
+    lasers = []
+    try:
+        for port in com_ports:
+            try:
+                laser = VoltranLaser(port=port.device)
+                if laser.serial_number != None:
+                    lasers.append((laser.port, laser.serial_number))
+                    logger.info(f"Found: {laser.port}:{laser.devID}")
+                    laser.disconnect()
+                else:
+                    raise Exception
+            except:
+                pass
+        if len(lasers) < 1:
+            raise Exception
+        return lasers
+    except:
+        raise RuntimeError("No lasers found...")
+
+
+class VoltranLaser(AbstractLight):
+    def __init__(self, devID=None, port=None, baudrate=19200, timeout=1):
```

Comment:
`device_id` could be a more explicit parameter name?

Reply:
changed in [04cffcc](https://github.com/czbiohub/coPylot/pull/148/commits/04cffcc53f2925aff1613e78e1b87f70d1fb7497)